oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/AheadOfTimeBlobDownloadingFlatFileStore.java

```diff
@@ -75,17 +80,35 @@ private AheadOfTimeBlobDownloadingFlatFileStore(FlatFileStore ffs, CompositeInde
                     maxPrefetchWindowSize,
                     maxPrefetchWindowMB);
         }
-    }    
+    }
+
+    static boolean isEnabledForIndexes(String indexesEnabledPrefix, List<String> indexPaths) {
+        List<String> enableForIndexes = splitAndTrim(indexesEnabledPrefix);
+        for (String indexPath : indexPaths) {
+            if (enableForIndexes.stream().anyMatch(indexPath::startsWith)) {
+                return true;
```

Comment:
... in which case we have "/oak:index/...".startsWith("")... that would always be true.

I think if enableForIndexes is empty, then this methods should always return false.

Reply:
There is a test case for when `indexesEnabledPrefix` is empty, the function correctly returns false. See the first case in the companion test class below. The loop won't be executed if the string is empty and it will fall through to the `return false` at the end.